webui/src/pages/repositories/repository/tags.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { useMemo, useRef, useState } from "react";
+
+import {
+    TagIcon,
+    LinkIcon,
+    PackageIcon,
+    TrashIcon
+} from "@primer/octicons-react";
+import ButtonGroup from "react-bootstrap/ButtonGroup";
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import ListGroup from "react-bootstrap/ListGroup";
+
+import {tags} from "../../../lib/api";
+
+import {
+    ActionGroup,
+    ActionsBar, ClipboardButton,
+    Error, LinkButton,
+    Loading, RefreshButton
+} from "../../../lib/components/controls";
+import { RepositoryPageLayout } from "../../../lib/components/repository/layout";
+import { RefContextProvider, useRefs } from "../../../lib/hooks/repo";
+import { useAPIWithPagination } from "../../../lib/hooks/api";
+import { Paginator } from "../../../lib/components/pagination";
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+import RefDropdown from "../../../lib/components/repository/refDropdown";
+import { Link } from "../../../lib/components/nav";
+import { useRouter } from "../../../lib/hooks/router";
+import {ConfirmationButton} from "../../../lib/components/modals";
+
+
+const TagWidget = ({ repo, tag, onDelete }) => {
```

Comment:
Empty tags screen looks missing, maybe state explicitly that it's empty?
![image](https://user-images.githubusercontent.com/9861474/140642461-aa9aa407-927a-477b-9d2f-4bddcefb4376.png)